### PR TITLE
Refactoring auth init and adding tests

### PR DESCRIPTION
**Pull request by cbb330**
## Summary
problem: 1) auth requirement is not documented in swagger UI 2) auth interceptor init is not tested

(for context, this PR is from a ticket that is meant for me to onboard to the team and learn the code. as part of skilling up I've used refactoring to learn the code better. also, a lot of this code was made in an attempt to add custom headers as types exposed in our client SDK which turned out to be more complexity than it is worth.)

changes:
- refactored client codegen shell scripts to use a parameterized openapi-codegen cli, unblocking version upgrade (bumped version to 6.6.0
- logic for converting from clusterproperties yaml -> java objects is pulled out from TablesMvc and into its own util class for testability, refactored for readability.
- - along with move and refactoring,  3 unittests to prevent regression
-  SwaggerConfig.java deleted and merged into TablesMvcConfigurer to both reduce # of places to look for config as well as utilize shared initialization steps. swagger will now init with security documented in UI if security is declared in cluster properties.
- Authorization is now required in e2e/h2/* tests which enables testing that the springboot app initializes properly with auth interceptor. new unittests for this 
- Automocked MVC will now include Authorization header by default, enabling unittests to use auth required endpoints and without any extra configuration
- moved AuthorizationPropertiesInitializer to be used in either h2 or mock locations. small unittest for this too 
- AuthorizationPropertiesInitializer now also defines auth interceptor

still TODO:

- [x] fix naming of getCons (handlerinterceptor class loader)
- [x] unittest the new HandlerLoader logic
- [x] add documentation in this PR
- [x] investigate whether front end docs + back end validation can be consolidated further

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [ ] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [x] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

added several unittests:
- covering clusterpropertiesutil 
- for proving interceptor initialization is correct and includes headers when needed
- some small tests for loading clusterproperties from yaml for use in unittests
- as well as deployed locally to verify that swagger had auth documented on endpoints

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: cluster/configs/src/main/java/com/linkedin/openhouse/cluster/configs/ClusterPropertiesUtil.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.linkedin.openhouse.cluster.configs;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Optional;
+import org.springframework.util.ClassUtils;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+public class ClusterPropertiesUtil {
```
Comment (cbb330) [inline cluster/configs/src/main/java/com/linkedin/openhouse/cluster/configs/ClusterPropertiesUtil.java:9]: this util is moved from tablesMVC configurer, to a generic and testable location.

File: services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java [hunk 2]
```diff
@@ -19,29 +23,45 @@
  */
 @Configuration
 public class TablesMvcConfigurer implements WebMvcConfigurer {
-
   @Autowired private ClusterProperties clusterProperties;
+  OpenAPI openAPI =
```
Comment (cbb330) [inline services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:27]: consolidating SwaggerConfig into this file, for benefit of having more centralized config. config across multiple files can lead to hard to trace keys/values when debugging. 

using this as inspiration: https://github.com/davidivkovic/uber-rides/blob/772317ce1c0baea8fd54f6e2b49c1a52e58a3853/src/services/rides/src/main/java/com/uber/rides/security/Config.java#L51

File: services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java [hunk 3]
```diff
@@ -19,29 +23,45 @@
...
+          .info(
+              new Info()
+                  .title("OpenHouse Tables APIs")
+                  .description("API description for OpenHouse Tables API")
+                  .termsOfService("http://swagger.io/terms")
+                  .version("v0.1")
+                  .license(new License().name("Apache 2.0").url("http://springdoc.org")));
 
   public void addInterceptors(InterceptorRegistry registry) {
-    if (clusterProperties.getClusterSecurityTokenInterceptorClassname() != null) {
```
Comment (cbb330) [inline services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:26]: moved out to its own util above

File: services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java [hunk 4]
```diff
@@ -19,29 +23,45 @@
...
-    }
+    Optional<HandlerInterceptor> handlerInterceptor =
+        ClusterPropertiesUtil.getClusterSecurityTokenInterceptor(clusterProperties);
+    handlerInterceptor.ifPresent(
+        interceptor -> {
+          registry
+              .addInterceptor(interceptor)
+              .addPathPatterns("/**")
+              .excludePathPatterns("/actuator/**", "/**/api-docs/**", "/**/swagger-ui/**");
+          addOpenAPISecurityScheme();
```
Comment (cbb330) [inline services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:46]: if handlerinterceptor is precent, do both interceptor and add security to openapi

File: services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java [hunk 5]
```diff
@@ -19,29 +23,45 @@
...
+        });
+  }
+
+  @Bean
+  public OpenAPI houseTablesOpenAPI() {
```
Comment (cbb330) [inline services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:51]: declare the bean available for springboot to pick up. 

the object here will have it's methods editted at startup by the addInterceptor method

File: services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java (hunk 2 above)
Comment (autumnust) [reply services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:27]: is there other services' configurer using this `SwaggerConfig` ?

Comment (autumnust) [reply services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:27]: ok I see this being annotated as a bean in the following. 
I believe the original intention to have SwaggerConfig separated is for sharing among different services

File: services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java [hunk 6]
```diff
@@ -19,29 +23,46 @@
...
+    return openAPI;
+  }
+
+  private void addOpenAPISecurityScheme() {
```
Comment (autumnust) [inline services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:55]: ok, this seems to be the core part of this PR.

**Review by autumnust — APPROVED** [review]
minor questions, otherwise LGTM

File: services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java (hunk 2 above)
Comment (cbb330) [reply services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:27]: there is no other service using this swagger config -- it is suited only for tables API / controller.

ideally we instantiate SwaggerConfig in global config. then Autowired it to each controller for extensions to e.g. the API info title ".title("OpenHouse Tables APIs") or .title("OpenHouse Jobs APIs")

but right now the APIs don't share anything in common so there is no benefit to converging them

File: services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java (hunk 6 above)
Comment (cbb330) [reply services/tables/src/main/java/com/linkedin/openhouse/tables/config/TablesMvcConfigurer.java:55]: yes, everything else was surrounding tests, refactoring to support tests, small tweaks to scripts, etc.

**Pull request merged**